yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -220,6 +248,13 @@ impl Parse for HtmlTagOpen {
         let TagSuffix { stream, div, gt } = input.parse()?;
         let mut attributes: TagAttributes = parse(stream)?;
 
+        if div.is_none() && VOID_ELEMENTS.contains(tag_name.to_string().as_str()) {
+            return Err(syn::Error::new_spanned(
+                gt,
+                &format!("<{}> tag must be self-closed", tag_name),
```

Comment:
I think that this error message could be improved. Something more like: `the tag <{}> is a void element and cannot have children (hint: rewrite this as <{}/>)`